File: tests/HartreeSinc/compute_F.jl
import SpecialFunctions

function compute_F( t::Float64, x_bar::Float64, scaling::Float64 )
    return_val = 0.0

    if x_bar < 1.e-30
        return_val= sqrt(scaling) * erf(pi/(2.0*scaling*t))
    else
        z = Complex128( pi/(2*scaling*t), t*x_bar )
        w_iz = erfcx(z)
        return_val = exp( -t*t*x_bar*x_bar )
        return_val = return_val - ( exp( -t*t*x_bar*x_bar - z*z ) * w_iz).re
        return_val = sqrt(scaling)*return_val
    end
    return return_val;
end

function test_compute_F()
     t = 0.1
     x_bar = 0.12
     scaling = 0.2

     F = compute_F(t, x_bar, scaling)

     @printf("F = %18.10f\n", F)
end

test_compute_F()

File: tests/HartreeSinc/compute_F.f90
FUNCTION compute_F( t, x_bar, h ) RESULT( f )
  USE m_constants, ONLY : PI, EPS_SMALL
  IMPLICIT NONE 
  REAL(8) :: t, x_bar, h
  REAL(8) :: f
  !
  COMPLEX(8) :: z, w_iz
  REAL(8) :: f_re, f_im

  f = 0.d0

  IF(x_bar < EPS_SMALL) THEN 
    f = sqrt(h) * erf(PI/(2.d0*h*t))
  ELSE 
    z = cmplx( PI/(2*h*t), t*x_bar, kind=8 )
    CALL Cwrap_faddeeva( real(z,kind=8), aimag(z), f_re, f_im )
    w_iz = cmplx( f_re, f_im, kind=8 )
    f = exp( -t*t*x_bar*x_bar )
    f = f - REAL( exp( -t*t*x_bar*x_bar - z*z )*w_iz, kind=8) 
    f = sqrt(h)*f
  ENDIF 
END FUNCTION 



